editors/code/src/installation/download_file.ts

```diff
@@ -0,0 +1,34 @@
+import fetch from "node-fetch";
+import { throttle } from "throttle-debounce";
+import * as fs from "fs";
+import { strict as assert } from "assert";
+
+/**
+ * Downloads file from `url` and stores it at `destFilePath`.
+ * `onProgress` callback is periodically called to track the progress of downloading,
+ * it gets the already read and total amount of bytes to read as its parameters.
+ */
+export async function downloadFile(
+    url: string,
+    destFilePath: fs.PathLike,
+    onProgress: (readBytes: number, totalBytes: number) => void
+): Promise<void> {
+    onProgress = throttle(200, /* noTrailing: */ true, onProgress);
```

Comment:
Seems like a more orthogonal and powerful interface would be to do throttling on the call site. Then the calling code can decide the delay and whether the throttling is requred at all.

Reply:
Hmm, yes, seems reasonable